src/plugins/dashboard/public/application/lib/diff_dashboard_state.ts

```diff
@@ -98,15 +107,67 @@ export const diffDashboardState = async ({
     newState.controlGroupInput
   );
 
+  const timeStatediff = getTimeSettingsAreEqual({
+    currentTimeRestore: newState.timeRestore,
+    lastSaved: { ...pick(originalState, ['timeRange', 'timeRestore', 'refreshInterval']) },
+  })
+    ? {}
+    : pick(newState, ['timeRange', 'timeRestore', 'refreshInterval']);
+
```

Comment:
Oooh! Much cleaner to loop in the time state difference stuff here instead of in `use_dashboard_app_state`. Nice 👍

Reply:
Amazing, thanks for re-reviewing!